great_expectations/datasource/fluent/databricks_sql_datasource.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, Union
+from urllib import parse
+
+import pydantic
+from pydantic import AnyUrl
+
+from great_expectations.compatibility.sqlalchemy import (
+    sqlalchemy as sa,
+)
+from great_expectations.core._docs_decorators import public_api
+from great_expectations.datasource.fluent.config_str import ConfigStr
+from great_expectations.datasource.fluent.sql_datasource import (
+    SQLDatasource,
+)
+
+if TYPE_CHECKING:
+    from pydantic.networks import Parts
+
+    from great_expectations.compatibility import sqlalchemy
+
+
+def _parse_http_path_from_query(query: str) -> str | None:
+    url_components = parse.urlparse(query)
+    path = str(url_components.path)
+    parse_results: dict[str, list[str]] = parse.parse_qs(path)
+    path_results = parse_results.get("http_path", [])
+
+    if path_results and len(path_results) == 1:
+        return path_results[0]
```

Comment:
```suggestion
    if path_results:
        return path_results[0]
```

This will silently fail if the user provides multiple `http_path` query entries.

I would suggest we either warn/fail or just select the first `http_path` entry. Not sure off the top of my head which is the best approach. 🤔

Reply:
the DSN has logic to raise an error if this is `None` - I shifted the burden of error handling to the callers of this util

Reply:
I don't think multiple `http_path` queries will be provided in a valid use-case (or maybe I'm wrong?)

Reply:
That makes sense. If only one `http_path` is allowed there we should add a simple check to fail if the user provides multiple `http_path` entries.
Query params can be repeated in a URI.

Reply:
done!